### PR TITLE
Smaato: Support floor price module

### PR DESCRIPTION
**Pull request by el-chuck**
# Type of change

- [x ] Feature
- [x ] Other

## Description of change

* Add support for floor price module
* Split multiple media types into multiple requests

LINK TO DOCS PR HERE

### CODE REVIEW

File: modules/smaatoBidAdapter.js [hunk 1]
```diff
@@ -388,11 +426,16 @@ const addOptionalAdpodParameters = (request, videoMediaType, imp) => {
...
+function getBidFloor(bidRequest, mediaType, sizes) {
+  if (typeof bidRequest.getFloor === 'function') {
+    const size = sizes.length === 1 ? sizes[0] : '*';
+    const floor = bidRequest.getFloor({currency: CURRENCY, mediaType: mediaType, size: size});
+    if (floor && !isNaN(floor.floor) && (floor.currency === CURRENCY)) {
+      return floor.floor;
+    }
+  }
+
+  return 0;
```
Comment (Enigo) [inline modules/smaatoBidAdapter.js:440]: shouldn't it be rather `null` (undefined) ? 0 bid floor doesn't make much sense

File: modules/smaatoBidAdapter.js [hunk 2]
```diff
@@ -388,11 +426,16 @@ const addOptionalAdpodParameters = (request, videoMediaType, imp) => {
...
-  imp.video.protocols = videoMediaType.protocols
-  imp.video.skipmin = videoMediaType.skipmin
-  imp.video.api = videoMediaType.api
+}
+
+function getBidFloor(bidRequest, mediaType, sizes) {
+  if (typeof bidRequest.getFloor === 'function') {
+    const size = sizes.length === 1 ? sizes[0] : '*';
+    const floor = bidRequest.getFloor({currency: CURRENCY, mediaType: mediaType, size: size});
+    if (floor && !isNaN(floor.floor) && (floor.currency === CURRENCY)) {
```
Comment (Enigo) [inline modules/smaatoBidAdapter.js:435]: should it mentioned anywhere (in docs) that we only support `USD`?

File: modules/smaatoBidAdapter.js [hunk 3]
```diff
@@ -388,11 +426,16 @@ const addOptionalAdpodParameters = (request, videoMediaType, imp) => {
...
-  imp.video.linearity = videoMediaType.linearity
-  imp.video.skip = videoMediaType.skip
-  imp.video.protocols = videoMediaType.protocols
-  imp.video.skipmin = videoMediaType.skipmin
-  imp.video.api = videoMediaType.api
+}
+
+function getBidFloor(bidRequest, mediaType, sizes) {
+  if (typeof bidRequest.getFloor === 'function') {
+    const size = sizes.length === 1 ? sizes[0] : '*';
```
Comment (Enigo) [inline modules/smaatoBidAdapter.js:433]: do we have a test case for `sizes.length != 1` ? cannot see a change for that but maybe it is covered already

File: modules/smaatoBidAdapter.js (hunk 2 above)
Comment (el-chuck) [reply modules/smaatoBidAdapter.js:435]: will update the doc changes

File: modules/smaatoBidAdapter.js (hunk 3 above)
Comment (el-chuck) [reply modules/smaatoBidAdapter.js:433]: brilliant :clap: 
added more tests, uncovered bugs, now fixed

File: test/spec/modules/smaatoBidAdapter_spec.js [hunk 4]
```diff
@@ -161,17 +161,20 @@ describe('smaatoBidAdapterTest', () => {
...
       });
 
       it('sends bidfloor when configured', () => {
         const singleBannerBidRequestWithFloor = Object.assign({}, singleBannerBidRequest);
-        singleBannerBidRequestWithFloor.getFloor = function() {
-          return {
-            currency: 'USD',
-            floor: 0.123
+        singleBannerBidRequestWithFloor.getFloor = function(arg) {
+          if (arg.currency === 'USD' &&
```
Comment (Enigo) [inline test/spec/modules/smaatoBidAdapter_spec.js:171]: well, I hope that having `if` statement in tests is okay practice in js XD

**Review by Enigo — COMMENTED** [review]
LGTM

File: test/spec/modules/smaatoBidAdapter_spec.js (hunk 4 above)
Comment (el-chuck) [reply test/spec/modules/smaatoBidAdapter_spec.js:171]: we will see :zap:

**Pull request closed without merging**